From dd0a5a06634137f4cc3b67bc48e6fa679b515d88 Mon Sep 17 00:00:00 2001
From: Vicente Adolfo Bolea Sanchez <vicente.bolea@kitware.com>
Date: Tue, 6 Sep 2022 20:07:13 -0400
Subject: [PATCH] HIP: Add PARAVIEW_USE_HIP option

---
 CMakeLists.txt | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 917127980a..3029dfcb4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -180,13 +180,24 @@ if (NOT DEFINED CMAKE_INSTALL_RPATH_USE_LINK_PATH)
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 endif ()
 
+cmake_dependent_option(PARAVIEW_USE_HIP "Support HIP backend accelerator" OFF
+  "CMAKE_VERSION VERSION_GREATER_EQUAL 3.21 AND NOT PARAVIEW_USE_CUDA" OFF)
+mark_as_advanced(PARAVIEW_USE_HIP)
+
+# The following `enable_language` needs to be a top level flag as
+# enable_language needs to be called from CMake entry point for variables to
+# have the correct scope.
 if(PARAVIEW_USE_CUDA)
-  # This needs to be a top level flag as enable_language needs
-  # to be called from CMake entry point for variables to have
-  # the correct scope.
   enable_language(CUDA)
 endif()
 
+if(PARAVIEW_USE_HIP)
+  if(VTK_USE_CUDA)
+    message(FATAL_ERROR "PARAVIEW_USE_HIP and PARAVIEW_USE_CUDA are mutually exclusive, enable only one.")
+  endif()
+  enable_language(HIP)
+endif()
+
 set(BUILD_TESTING OFF)
 if (PARAVIEW_BUILD_TESTING)
   set(BUILD_TESTING ON)
@@ -469,6 +480,8 @@ else ()
 
   # Forward CUDA option to VTK
   set(VTK_USE_CUDA "${PARAVIEW_USE_CUDA}")
+  # Forward HIP option to VTK
+  set(VTK_USE_HIP ${PARAVIEW_USE_HIP})
   # Forward Memkind option to VTK.
   set(VTK_USE_MEMKIND "${PARAVIEW_USE_MEMKIND}")
   # Forward OpenVDB option to VTK.
-- 
GitLab

